### DISCUSSION TITLE
Best Ccache options for Xcode compiling

### DISCUSSION THREAD

**Discussion by adamivancza** [ccache/ccache#1405] [Questions & answers]
I have the following options set for ccache settings for our react-native iOS project:

```
  export CCACHE_MAXSIZE=10G
  export CCACHE_CPP2=true
  export CCACHE_DIR=~/ccache
  export CCACHE_SLOPPINESS=pch_defines,time_macros,file_macro,include_file_mtime,include_file_ctime,file_stat_matches,modules,clang_index_store,system_headers,ivfsoverlay
  export CCACHE_DEPEND=true
  export CCACHE_FILECLONE=true
  export CCACHE_INODECACHE=true
  export CCACHE_COMPILERCHECK=content
```

but our cache hit ratio is still relatively small:

```
Cacheable calls:    13020 / 13054 (99.74%)
  Hits:              3254 / 13020 (24.99%)
    Direct:          3254 /  3254 (100.0%)
    Preprocessed:       0 /  3254 ( 0.00%)
  Misses:            9766 / 13020 (75.01%)
Uncacheable calls:     34 / 13054 ( 0.26%)
Local storage:
  Cache size (GiB):   2.1 /   5.0 (42.39%)
  Hits:              3254 / 13020 (24.99%)
  Misses:            9766 / 13020 (75.01%)
```

Any suggestions what options should I change to improve the cache hit ratio? There was 0 native changes between when the cache was generated and between the build that produced the above cache hit ratios.

**Comment by jrosdahl** [comment]
To find out why you don't get hits I suggest following the hint at the end of [_Cache debugging_](https://ccache.dev/manual/4.9.1.html#_cache_debugging) in the manual.

**Comment by biodranik** [comment]
> clang_index_store

> Ignore the Clang compiler option -index-store-path and its argument when computing the manifest hash. This is useful if you use Xcode, which uses an index store path derived from the local project path. Note that the index store won’t be updated correctly on cache hits if you enable this sloppiness.

> ivfsoverlay

> Ignore the Clang compiler option -ivfsoverlay and its argument. This is useful if you use Xcode, which uses a virtual file system (VFS) for things like combining Objective-C and Swift code.


https://ccache.dev/manual/latest.html
